### ISSUE TITLE
[FEATURE] Add full text search to lookup files related to text search

### ISSUE THREAD

**Issue by prikeshsavla** [haxzie/snipp.in#13]
Add fulltext search functionality to search in already created files like sublime or vscode.

**Comment by haxzie** [comment]
This would be a cool addition. I was thinking this would need a couple of UI changes too.

**Comment by prikeshsavla** [comment]
Yeah, you can have a modal open on shortcut like sublime or jetbrains  or keep the search in the side bar.

**Comment by prikeshsavla** [comment]
I figure the searching in files code with regex. I might need to look at more combinations or a more intelligent regex to lookup properly with the keywords. the biggest issue I faced that it loaded the whole set in a variable before filtering through it. Which seems excessive in terms of performance. I can share a snippet you guys can direct a better solution maybe. 


```
  searchForFiles: async ({ state, dispatch }, { query }) => {
    const searchResults = await db.files
      .filter((file) => {
        return `/${query}/i`.test(file.name) || `/${query}/i`.test(file.contents);
      }).toArray();
      return searchResults;
  },
```

**Comment by prikeshsavla** [comment]
Or Is it good to use a external library altogether like  [FuseJs](https://fusejs.io)? 

https://fusejs.io/examples.html#weighted-search

**Comment by prikeshsavla** [comment]
I used FuseJS in another side project I am working on for nuxt. It works fine. I can put a pr for a store action which can return search results.

**Comment by FlorianWoelki** [comment]
Yeah, FuseJS is beautiful :) I would say, go for it 👍

**Comment by prikeshsavla** [comment]
How does this look. I will need some refactoring an code review from you guys for UI and code. 

![captured](https://user-images.githubusercontent.com/34878926/106995221-72028f80-67a4-11eb-863e-31fc2990bb8c.gif)

**Comment by prikeshsavla** [comment]
#28

**Comment by haxzie** [comment]
Implemented via #34 :+1:
